### PR TITLE
Claim storage

### PR DESCRIPTION
**Pull request by Justincjr**

### CODE REVIEW

File: src/main/java/seedu/address/model/policy/Policy.java [hunk 1]
```diff
@@ -27,27 +28,28 @@ public abstract class Policy {
      * @param expiryDate the date of Policy's expiry.
      * @throws NullPointerException if any of the fields is null.
      */
-    public Policy(PremiumAmount premiumAmount, CoverageAmount coverageAmount, ExpiryDate expiryDate) {
+    public Policy(PremiumAmount premiumAmount, CoverageAmount coverageAmount, ExpiryDate expiryDate,
+                  ClaimList claims) {
         requireAllNonNull(premiumAmount, coverageAmount, expiryDate);
```
Comment (yihuididi) [inline src/main/java/seedu/address/model/policy/Policy.java:33]: add claims to requireAllNonNull

File: src/main/java/seedu/address/model/policy/EducationPolicy.java [hunk 2]
```diff
@@ -30,11 +33,13 @@ public EducationPolicy() {
...
      */
-    public EducationPolicy(PremiumAmount premiumAmount, CoverageAmount coverageAmount, ExpiryDate expiryDate) {
+    public EducationPolicy(PremiumAmount premiumAmount, CoverageAmount coverageAmount, ExpiryDate expiryDate,
+                           ClaimList claims) {
         super(
                 premiumAmount == null ? DEFAULT_PREMIUM_AMOUNT : premiumAmount,
                 coverageAmount == null ? DEFAULT_COVERAGE_AMOUNT : coverageAmount,
-                expiryDate == null ? new ExpiryDate(LocalDate.now().plus(DEFAULT_EXPIRY_DATE_PERIOD)) : expiryDate);
+                expiryDate == null ? new ExpiryDate(LocalDate.now().plus(DEFAULT_EXPIRY_DATE_PERIOD)) : expiryDate,
+                claims);
```
Comment (yihuididi) [inline src/main/java/seedu/address/model/policy/EducationPolicy.java:42]: should check if claims is null. If it is null, instantiate an empty claimlist. Same for Health and Lifepolicy.

File: src/main/java/seedu/address/storage/JsonAdaptedPolicy.java [hunk 3]
```diff
@@ -25,11 +34,13 @@ class JsonAdaptedPolicy {
...
                              @JsonProperty("premiumAmount") double premiumAmount,
                              @JsonProperty("coverageAmount") double coverageAmount,
-                             @JsonProperty("expiryDate") String expiryDate) {
+                             @JsonProperty("expiryDate") String expiryDate,
+                             @JsonProperty("claims")List<JsonAdaptedClaim> claims) {
         this.policyType = policyType;
         this.premiumAmount = premiumAmount;
         this.coverageAmount = coverageAmount;
         this.expiryDate = expiryDate;
+        this.claims.addAll(claims);
```
Comment (yihuididi) [inline src/main/java/seedu/address/storage/JsonAdaptedPolicy.java:43]: do if (claims != null) like in JsonAdaptedPerson

File: src/main/java/seedu/address/logic/parser/AddPolicyCommandParser.java [hunk 4]
```diff
@@ -57,7 +58,7 @@ public AddPolicyCommand parse(String args) throws ParseException {
         ExpiryDate expiryDate = ParserUtil.parseExpiryDate(
                 argMultimap.getValue(PREFIX_POLICY_EXPIRY_DATE).orElse(""));
 
-        Policy policy = Policy.makePolicy(policyType, premiumAmount, coverageAmount, expiryDate);
+        Policy policy = Policy.makePolicy(policyType, premiumAmount, coverageAmount, expiryDate, new ClaimList());
```
Comment (yihuididi) [inline src/main/java/seedu/address/logic/parser/AddPolicyCommandParser.java:61]: use null for claimList. The respective constructors for LifePolicy, HealthPolicy, and EducationPolicy will check for null inputs and create the new empty ClaimList. Do the same for all instance where you called makePolicy with an empty ClaimList.

**Review by ckaayy — APPROVED** [review]
lgtm!

**Pull request merged**